immuni_exposure_ingestion/tasks/process_uploads.py

```diff
@@ -60,70 +61,145 @@ async def _process_uploads() -> None:
-            # Sort the keys. This randomizes their order (since they are random strings) so that
-            # keys of the same device are no more likely to end up consecutively.
-            keys = sorted(keys, key=lambda x: x.key_data)
-
-            index = last_index + 1
-
-            batch_file = BatchFile(
-                index=index,
-                keys=keys,
-                period_start=period_start,
-                period_end=period_end,
-                sub_batch_index=1,
-                sub_batch_count=1,
+    uploads = Upload.to_process()
+
+    _LOGGER.info("Uploads have been fetched.", extra=dict(n_uploads=uploads.count()))
+
+    processed_uploads: List[ObjectId] = []
+    keys: List[TemporaryExposureKey] = []
+    for upload in uploads:
+        if (reached := len(keys) + len(upload.keys)) > config.MAX_KEYS_PER_BATCH:
+            _LOGGER.warning(
+                "Early stop: reached maximum number of keys per batch.",
+                extra=dict(
+                    pre_reached=len(keys), reached=reached, max=config.MAX_KEYS_PER_BATCH
+                ),
             )
-            batch_file.client_content = batch_to_sdk_zip_file(batch_file)
-            batch_file.save()
-            _LOGGER.info("Created new batch.", extra=dict(index=index, n_keys=n_keys))
-            BATCH_FILES_CREATED.inc()
-            KEYS_PROCESSED.inc(len(keys))
-
-        Upload.set_published(processed_uploads)
-        _LOGGER.info(
-            "Flagged uploads as published.", extra=dict(n_processed_uploads=len(processed_uploads))
+            break
+        keys += extract_keys_with_risk_level_from_upload(upload)
+        processed_uploads.append(upload.id)
+
+    if (n_keys := len(keys)) > 0:
+        # Sort the keys. This randomizes their order (since they are random strings) so that
+        # keys of the same device are no more likely to end up consecutively.
+        keys = sorted(keys, key=lambda x: x.key_data)
+
+        index = last_index + 1
+
+        batch_file = BatchFile(
+            index=index,
+            keys=keys,
+            period_start=period_start,
+            period_end=period_end,
+            sub_batch_index=1,
+            sub_batch_count=1,
+            origin="IT"
         )
-        UPLOADS_ENQUEUED.set(Upload.to_process().count())
-        _LOGGER.info("Releasing lock.")
+        batch_file.client_content = batch_to_sdk_zip_file(batch_file)
+        batch_file.save()
+        _LOGGER.info("Created new batch.", extra=dict(index=index, n_keys=n_keys))
+        BATCH_FILES_CREATED.inc()
+        KEYS_PROCESSED.inc(len(keys))
 
-    _LOGGER.info("Upload processing completed successfully.")
+    Upload.set_published(processed_uploads)
+    _LOGGER.info(
+        "Flagged uploads as published.", extra=dict(n_processed_uploads=len(processed_uploads))
+    )
+    UPLOADS_ENQUEUED.set(Upload.to_process().count())
+
+
+def batch_eu():
+
+    infos = BatchFile.get_latest_info()
```

Comment:
1. This should be BatchFileEU, right?
2. Since the logic is the same, I'd keep a single method, whose interface is the batch and upload classes
```python
_process_uploads(
    batch_class: Union[Type[BatchFile], Type[BatchFileEU]],
    upload_class: Union[Type[Upload], Type[UploadEU]]
)
```

Reply:
No, because as said in the previous comment, in this function we process the foreign Teks marked as italian to be sent to all Italian citizens according to the current flow.

Reply:
I'll re-read this tomorrow. I want to better understand its implications.